src/engine/MaxConstraint.cpp

```diff
@@ -712,3 +777,35 @@ bool MaxConstraint::haveOutOfBoundVariables() const
     }
     return false;
 }
+
+void MaxConstraint::createElementTighteningRow( unsigned element )
+{
+
+    // Create the row only when needed and when not already create
+    if ( !_boundManager->getBoundExplainer() || _elementToTighteningRow[element] != NULL )
+        return;
+
```

Comment:
I don't understand the logic here. Aren't tightenings performed according to existing rows, or according to contraint lemmas? Why do we need to create a row, then?

Reply:
They are. However when we update an explanation according to a tableau row, we need to find and access it. The tableau functions require either heavy computation, or would just copy the same row, so I think it is more elegant this way.
We do it too in the existing proof production for ReLU.